code/modules/core_implant/cruciform/machinery/cruciformforge.dm

```diff
@@ -0,0 +1,147 @@
+#define LOAD "load"
+#define WORK "work"
+#define DONE "done"
+
+/obj/machinery/neotheology/cruciformforge
+	name = "Cruciform Forge"
+	desc = "This unique piece of technology can be fed biomatter, plasteel, and gold in order to print more cruciforms. Only a select few NeoTheologian architects know how to create these machines. "
+	icon = 'icons/obj/neotheology_machinery.dmi'
+	icon_state = "cruciforge"
+
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
+
+	var/power_cost = 250
+
+	var/working = FALSE
+	var/start_working
+	var/work_time = 30 SECONDS
+	var/storage_capacity = 50
+	var/list/stored_material = list()
+	var/list/needed_material = list(MATERIAL_BIOMATTER = 10, MATERIAL_PLASTEEL = 5, MATERIAL_GOLD = 2)
+	var/spawn_type = /obj/item/weapon/implant/core_implant/cruciform
+
+	// A vis_contents hack for materials loading animation.
+	var/tmp/obj/effect/flicker_overlay/image_load
+
+/obj/machinery/neotheology/cruciformforge/Initialize()
+	. = ..()
+
+	image_load = new(src)
+
+	for(var/_material in needed_material)
+		stored_material[_material] = rand(1, 10)
```

Comment:
Why not just put the necessary materials on a table next to the machine?

Reply:
There is a tiny space for that. Also, it gives the same feeling as it was used before.